pkg/api/conversions/conversions.go

```diff
@@ -0,0 +1,95 @@
+package migrate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	machinev1alpha1downstream "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(in machinev1alpha1downstream.Machine) (*clusterv1alpha1.Machine, error) {
```

Comment:
we could stick to how k8s defines conversion function, for example this is the method that will convert `StatefulSetSpec`

`Convert_v1beta1_StatefulSetSpec_To_apps_StatefulSetSpec(in *appsv1beta1.StatefulSetSpec, out *apps.StatefulSetSpec, s conversion.Scope) error`

so we could change our function to have the following signature ?
`Convert_v1alpha1_DownStreamMachine_To_v1alpha1_ClusterMachine(in *machinev1alpha1downstream.Machine, out *clusterv1alpha1.Machine) error`

If we don't like it then perhaps we could change our signature to accept and return a pointer to a struct ?
`func ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(in *machinev1alpha1downstream.Machine) (*clusterv1alpha1.Machine, error)`